qa: parametrize logger usages

jdrueckert opened the pull request:
Re-does MovingBlocks/Terasology#5173 (reverting its revert MovingBlocks/Terasology#5180)

jdrueckert committed:
Revert "revert: "qa: parametrize logger usages (#5173)" (#5180)"

This reverts commit 4f42a8fa1a17e4b432edb1c23e27177c10dd58db.

jdrueckert commented:
@soloturn let's split out the slf4j upgrade and any lazy log adjustments
thus, the remaining changes can be merged until we get the slf4j upgrade to work

soloturn committed:
revert upgrade slf4j-2.0

soloturn commented:
how do you want me to get a commit in here? like this #5182 ?

jdrueckert commented:
> how do you want me to get a commit in here? like this #5182 ?

you _should_ have permissions to work on branches on the original repo, not your fork 🤔
this _should_ also allow you to checkout and commit on an existing branch...

if this is not the case, please tell me so I can recheck your permissions

the idea would be to remove the slf4j upgrade from this branch and open a new PR (targeting `develop`) with the removed changes

soloturn commented:
no right
```
❯ git push
ERROR: Permission to MovingBlocks/Terasology.git denied to soloturn.
fatal: Could not read from remote repository.

```

soloturn committed:
micrometer 1.9.8 --> 1.9.12

jdrueckert commented:
@soloturn you should've received an invitation to the `ReviverWriteAccess` GitHub team
Becoming a member of that team provides you with privileges to push branches and commits to the MovingBlocks/Terasology and MovingBlocks/TeraConfig repos as well as manage issues and PRs.

soloturn commented:
redone, there is one line less than in the original, failing one. engine/build.gradle stays on slf4j-1.7.36, @jdrueckert. for details see 7f4e40f06633b980d454c7c87de796572a09b8d9 and https://github.com/MovingBlocks/Terasology/pull/5182.

jdrueckert commented on build-logic/src/main/kotlin/terasology-metrics.gradle.kts:
```diff
@@ -25,13 +25,13 @@ dependencies {
         because("allows use of EvaluatorFilter in logback.xml")
     }
-    testRuntimeOnly("org.slf4j:jul-to-slf4j:1.7.36") {
+    testRuntimeOnly("org.slf4j:jul-to-slf4j:2.0.9") {
```
@soloturn I think we should drop the upgrade from this PR entirely, not only for engine.
Thus, we can update it for all parts simultaneously.

soloturn replied:
i always agree with your judgement. ok, most of the times :) but here i do not agree - using different jars or versions for testing must be possible, otherwise something is seriously wrong in the dependency tree, @jdrueckert . what am i overlooking ?

jdrueckert replied:
@soloturn I'm not saying that it isn't possible to use different versions :smile: 
I'm saying: if we noticed issues with slf4j v2, we should consider sticking with 1.7 for our tests as well until we figured out the issues with v2 at which point we can update it for engine and engine tests.

soloturn replied:
@jdrueckert perfect, if we can use different versions, thats all i wanted :smile:  we know exaclty where and why it breaks, at one point. reason is in the hs_err_pid65260.log, in #5182. engine contains code relying on the old version, RegexFilterAction. jvm does not find the old binding, and crashes.

soloturn replied:
@jdrueckert tested, and it is indeed not even used. so remove. also reverted back jul-to-slf4j.

soloturn committed:
qa, still use old slf4j, feedback jdrueckert.

gradle does not care about the logger definition in engine-tests, it
uses the one from engine also for the tests. remove it to not confuse.

jdrueckert commented on engine-tests/build.gradle:
```diff
@@ -62,11 +62,7 @@ dependencies {
 
     implementation("org.terasology.joml-ext:joml-test:0.1.0")
 
-    implementation('org.slf4j:slf4j-api:1.7.36') {
-        because('a backend-independent Logger')
-    }
```
@soloturn why are we removing this here?

jdrueckert replied:
only saw your comment now...

>  tested, and it is indeed not even used. so remove. also reverted back jul-to-slf4j.

so you're saying we don't use slf4j for our test stuff? interesting... what's used for logging in tests instead then? :thinking: 
I thought I saw logs in some tests, but maybe I misremember?

soloturn replied:
slf4j comes into ```engine-tests``` via ```engine``` whose version wins. a lot of tohers bring slf4j in differing version as well, like gestalt, jnbullet, nui, nui-reflect, nui-gestatl, etc etc says ```gralde -q dependencies ``` in folder ```engine-tests```, in both compile and runtime classpath. i will be more explicit in the commit comment next time, @jdrueckert .

jdrueckert committed:
Merge branch 'develop' into revert-5180-revert-5173-qa/logger

jdrueckert committed:
Merge branch 'develop' into revert-5180-revert-5173-qa/logger

BenjaminAmos reviewed (approved):
These changes look mostly stylistic rather than functional. The dependency upgrades seem minor and ideally shouldn't break anything. There shouldn't be much harm in merging this now.
